presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/FileBasedAuthenticatorFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+import io.airlift.bootstrap.Bootstrap;
+import io.airlift.log.Logger;
+import io.airlift.units.Duration;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.security.PasswordAuthenticator;
+import io.prestosql.spi.security.PasswordAuthenticatorFactory;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Suppliers.memoizeWithExpiration;
+import static com.google.common.base.Throwables.throwIfUnchecked;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.prestosql.plugin.password.filebasedauthenticator.FileBasedAuthenticationConfig.FILEAUTH_REFRESH_PERIOD;
+import static io.prestosql.spi.StandardErrorCode.CONFIGURATION_INVALID;
+import static java.lang.String.format;
+
+public class FileBasedAuthenticatorFactory
+        implements PasswordAuthenticatorFactory
+{
+    private static final Logger log = Logger.get(FileBasedAuthenticatorFactory.class);
+
+    @Override
+    public String getName()
+    {
+        return "file";
+    }
+
+    @Override
+    public PasswordAuthenticator create(Map<String, String> config)
+    {
+        String configFileName = getName();
+        if (config.containsKey(FILEAUTH_REFRESH_PERIOD)) {
+            Duration refreshPeriod;
+            try {
+                refreshPeriod = Duration.valueOf(config.get(FILEAUTH_REFRESH_PERIOD));
+            }
+            catch (IllegalArgumentException e) {
+                throw invalidRefreshPeriodException(config);
+            }
+            if (refreshPeriod.toMillis() == 0) {
```

Comment:
this is proper value, you should not just not cache here.

Reply:
Changed.